File: server/python/src/metaobjects/render/prompt/prompt_overrides.py
"""FR-010 artifact 1 — render-time overrides of the metadata defaults."""
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Final

from metaobjects.render.prompt.prompt_style import PromptStyle


@dataclass(frozen=True, slots=True)
class PromptOverrides:
    """Render-time overrides of the metadata defaults.

    ``style`` of ``None`` keeps the spec's style; the maps override
    ``PromptField.example``/``PromptField.instruction`` per field name.
    """

    style: PromptStyle | None = None
    examples: dict[str, str] = field(default_factory=dict)
    instructions: dict[str, str] = field(default_factory=dict)


# No overrides — keep every metadata default. Mirrors Java ``PromptOverrides.none()``.
PROMPT_OVERRIDES_NONE: Final = PromptOverrides()


def no_overrides() -> PromptOverrides:
    """No overrides — keep every metadata default."""
    return PROMPT_OVERRIDES_NONE

File: server/typescript/packages/render/src/prompt/prompt-overrides.ts
// FR-010 artifact 1 — render-time overrides of the metadata defaults.

import type { PromptStyle } from "./prompt-style.js";

/**
 * Render-time overrides of the metadata defaults. `style` undefined keeps the
 * spec's style; the maps override `PromptField.example`/`PromptField.instruction`
 * per field name.
 */
export interface PromptOverrides {
  readonly style?: PromptStyle;
  readonly examples?: Readonly<Record<string, string>>;
  readonly instructions?: Readonly<Record<string, string>>;
}

/** No overrides — keep every metadata default. Mirrors Java `PromptOverrides.none()`. */
export const PROMPT_OVERRIDES_NONE: PromptOverrides = Object.freeze({
  // `style` omitted (not `undefined`) under exactOptionalPropertyTypes: an absent
  // style keeps the spec's own style, matching Java `PromptOverrides.none()`.
  examples: {},
  instructions: {},
});

/** No overrides — keep every metadata default. */
export function noOverrides(): PromptOverrides {
  return PROMPT_OVERRIDES_NONE;
}
